app/pages/Pdf/ServicePdfPage.tsx

```diff
@@ -0,0 +1,421 @@
+            padding: 0px 26px;
+          }
+
+          .renderedMarkdown ol li {
+            list-style-type: decimal;
+          }
+
+          .renderedMarkdown p strong {
+            color: #30383a;
+          }
+
+          .renderedMarkdown h1 {
+            font-size: 16px;
+            font-family: "Open Sans", "San Francisco", "Roboto", "Arial", sans-serif;
+          }
+
+          .locationSection {
+            padding-top: 9px;
+          }
+
+          .locationSection h2 {
+            text-transform: uppercase;
+            font-size: 18px;
+            font-weight: 800;
+            color: #484848;
+            padding-bottom: 6px;
+          }
+
+          .locationContainer {
+            display: grid;
+            grid-template-columns: auto 6fr;
+            gap: 9px;
+            align-items: center;
+            justify-items: end;
+          }
+
+          .resourceName {
+            font-weight: 600;
+          }
+
+          .serviceHours {
+            padding-top: 6px;
+          }
+
+          .serviceHours table.compact tr th,
+          .serviceHours table.compact tr td {
+            padding-bottom: 2px;
+            padding-top: 6px;
+          }
+
+          .serviceHours table tr th {
+            font-weight: 600;
+          }
+
+          .hoursTitle {
+            font-weight: 600;
+            text-decoration: underline;
+          }
+
+          .serviceHandoutPdf .map {
+            height: 300px;
+          }
+
+        `}
+          </style>
+          <header className="titleSection">
+            <h1 className="serviceTitle">{service.name}</h1>
+            <ServiceProgramDetails service={service} organization={resource} />
+          </header>
+
+          <ServiceListingSection title="About" className="aboutSection">
+            <ReactMarkdown
+              className="renderedMarkdown"
+              source={service.long_description}
+              linkTarget="_blank"
+            />
+            <ServiceAttribution
+              attribution={resource.source_attribution}
+              status={resource.status}
+            />
+          </ServiceListingSection>
+
+          {details.length > 0 && (
+            <ServiceListingSection className="detailsSection">
+              <Datatable
+                rowRenderer={(d) => (
+                  <>
+                    <tr key={d.title}>
+                      <th>{d.title}</th>
+                    </tr>
+                    <tr>
+                      <td>
+                        <ReactMarkdown className="renderedMarkdown">
+                          {d.value}
+                        </ReactMarkdown>
+                      </td>
+                    </tr>
+                  </>
+                )}
+                rows={details}
+              />
+            </ServiceListingSection>
+          )}
+
+          <ServiceListingSection className="contactInfo" title="Contact Info">
+            <TableOfContactInfo service={service} />
```

Comment:
I have maintenance concerns about using an inline stylesheet in this file while reusing the components from elsewhere in the app. When using CSS modules, I generally think of the class names and the HTML structure of the components as being "private" to the component, and therefore as a developer working on those components, I would feel free to change the internal HTML structure or CSS styles without having to check other pages.

That would be a very bad assumption of mine to make for any components that we're reusing here, since we're directly reaching into the internal structure with our inline stylesheet on this page, and any changes in the future that is done to these components could completely break the styling of this PDF.

Is there a way for us to either mitigate the pitfall of forgetting to update or even check the PDFs when making changes to the shared components in the future or to better isolate the PDF page from the internal details of the shared components?

Or is there even a way to inline the actual site's CSS into this HTML?

I would almost even suggest that we completely copy and paste the other components into this file so that there is absolutely zero sharing. This is obviously not DRY, but I feel like the danger of accidentally breaking the PDF with no way to easily automatically check them is greater than any drawbacks to duplicating code.

Perhaps even adding a very large comment to all of the components that you've imported here, letting devs know in the future that if they modify any of the HTML structure of those components, they should check that the CSS selectors on this PDF page still work correctly.

I'm not sure what the best solution is here, since I've never had to work on an app that needed a separate inline stylesheet for components that are shared with normal web HTML pages.

Reply:
> The constraint of needing to use inline styles for the PDF generation is interesting, and I have some concerns about it, but I'm not sure what the right solution is. I'm posting my comments now, but I'll sleep on it, and maybe we should chat about it sometime later.
> 
> Just as a reminder, I'm going to be on vacation without my laptop for the first half of February, so if you need to get this merged and deployed, then don't wait on my approval or anything, but I think it's an interesting technical challenge to solve.

Thanks for reviewing this, Richard! It was a hard one to review for a variety of reasons, I imagine.

In the end, I decided to just copy over the external components that this file references, so that future developers can safely edit those components and it won't affect this component's UI. Most if not all of the components that this file had been referencing (all of which are also referenced by the Service Listing Page), don't even use CSS modules. Instead they get their styles from a global style sheet `/app/styles/components/_listing-page.scss`, which uses element tags for the most part, and some classes, for defining styles (thus the element table styles I had to set as overrides inline).  It is not DRY to do this, but I am not too bothered by this since I think this component is practicaly presentation only and not likely to be referenced elsewhere.

I pasted the copied components at the end of this file and made comments that they'd been copied over. As a bonus this better enables us to use classes for styling since we can edit these components as we please without affecting other sections of the code. I have added some classes to the components, but not comprehensively in the interest of time. I will create a ticket for cleaning up the selectors for this file, which I think will be great for a beginner.

Let me know what you think and/or if you have other concerns. Thanks!